extensions/store/fcc-store-sql/src/main/java/de/truzzt/edc/extension/catalog/cache/sql/schema/BaseSqlDialectStatements.java

```diff
@@ -0,0 +1,71 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package de.truzzt.edc.extension.catalog.cache.sql.schema;
+
+import de.truzzt.edc.extension.catalog.cache.sql.schema.postgres.ContractOfferMapping;
+import org.eclipse.edc.spi.query.QuerySpec;
+import org.eclipse.edc.sql.translation.SqlQueryStatement;
+
+import static java.lang.String.format;
+
+public class BaseSqlDialectStatements implements ContractOfferStatements {
+
+    @Override
+    public String getInsertTemplate() {
+        return format("INSERT INTO %s (%s, %s, %s, %s, %s, %s, %s, %s, %s) VALUES (?, ?%s, ?, ?, ?, ?, ?, ?, ?)",
+                getContractOfferTable(),
+                getIdColumn(),
+                getPolicyColumn(),
+                getAssetIdColumn(),
+                getUriProviderColumn(),
+                getUriConsumerColumn(),
+                getOfferStartColumn(),
+                getOfferEndColumn(),
+                getContractStartColumn(),
+                getContractEndColumn(),
+                getFormatAsJsonOperator()
+        );
+    }
+
+    @Override
+    public String getCountTemplate() {
+        return format("SELECT COUNT (%s) FROM %s WHERE %s = ?",
+                getIdColumn(),
+                getContractOfferTable(),
+                getIdColumn());
+    }
+
+    public String getUpdateOfferEndTemplate() {
+        return format("UPDATE %s SET %s = ? WHERE %s IS NULL",
+                getContractOfferTable(),
+                getOfferEndColumn(),
+                getOfferEndColumn());
+    }
```

Comment:
It expires all the offers, which don't have a `offerEnd` set. I personally like this behavior, but it could be a different behavior as intended...
For now, we should definitely put a comment somewhere, because if someone expects to clean all offers with `expireAll()` and `deleteExpired()` it will not work as expected..